### PR TITLE
find cppcheck on windows

### PR DESCRIPTION
**Pull request by dirk-thomas**
@esteve @tfoote @wjwwood Please review.

### CODE REVIEW

**Comment by tfoote** [comment]
lgtm

File: ament_cppcheck/bin/ament_cppcheck [hunk 1]
```diff
@@ -41,7 +42,11 @@ def main(argv=sys.argv[1:]):
         print('No files found', file=sys.stderr)
         return 1
 
-    cppcheck_bin = find_executable('cppcheck')
+    additional_paths = None
+    if os.name == 'nt':
+        # search in location where cppcheck is installed via chocolatey
+        additional_paths = ['C:/Program Files (x86)/Cppcheck']
```
Comment (esteve) [inline ament_cppcheck/bin/ament_cppcheck]: Please use `%ProgramFiles%` instead. `Program Files` is mapped to a localized path in non-English installations of Windows:

http://en.wikipedia.org/wiki/Program_Files

**Comment by wjwwood** [comment]
+1

File: ament_cppcheck/bin/ament_cppcheck (hunk 1 above)
Comment (wjwwood) [inline ament_cppcheck/bin/ament_cppcheck]: It would be `%ProgramFiles(x86)%` or both.

Comment (wjwwood) [inline ament_cppcheck/bin/ament_cppcheck]: The problem is that I don't think that will get expanded by the `which` command, instead you'll need to get `program_file = os.environ.get('ProgramFiles(x86)', "C:/Program Files (x86)/")` and use that in the extra path.

Comment (dirk-thomas) [inline ament_cppcheck/bin/ament_cppcheck]: Does the environment variable contain a trailing slash?

Comment (wjwwood) [inline ament_cppcheck/bin/ament_cppcheck]: Doesn't look like it:

screen shot 2015-04-06 at 3 18 22 pm

File: ament_cppcheck/bin/ament_cppcheck [hunk 2]
```diff
@@ -41,7 +42,12 @@ def main(argv=sys.argv[1:]):
         print('No files found', file=sys.stderr)
         return 1
 
-    cppcheck_bin = find_executable('cppcheck')
+    additional_paths = None
+    if os.name == 'nt':
+        # search in location where cppcheck is installed via chocolatey
+        program_files_32 = os.environ.get('ProgramFiles(x86)', 'C:/Program Files (x86)')
```
Comment (wjwwood) [inline ament_cppcheck/bin/ament_cppcheck]: `\` rather than `/`

File: ament_cppcheck/bin/ament_cppcheck [hunk 3]
```diff
@@ -41,7 +42,12 @@ def main(argv=sys.argv[1:]):
...
+        additional_paths = [os.path.joni(program_files_32, 'Cppcheck')]
```
Comment (wjwwood) [inline ament_cppcheck/bin/ament_cppcheck]: `join` not `joni`.

**Comment by wjwwood** [comment]
Other than the comments +1

**Comment by esteve** [comment]
+1

**Comment by dirk-thomas** [comment]
Finally passed on the farm: http://54.183.26.131:8080/view/ros2/job/ros2_batch_ci_windows_cppcheck/8/

**Pull request merged**